components/file_system/src/thread_io.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::IOBytes;
+use crate::IOType;
+use std::fs::File;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::MutexGuard;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+use nix::unistd::getpid;
+use nix::unistd::gettid;
+use nix::unistd::Pid;
+use strum::EnumCount;
+use thread_local::ThreadLocal;
+
+lazy_static! {
+    static ref THREAD_IO_SENTINEL_VEC: Mutex<ThreadLocal<Arc<Mutex<ThreadIOSentinel>>>> =
+        Mutex::new(ThreadLocal::new());
+}
+
+thread_local! {
+    static THREAD_IO_SENTINEL: Arc<Mutex<ThreadIOSentinel>> = THREAD_IO_SENTINEL_VEC.lock().unwrap().get_or(||
+        Arc::new(Mutex::new(ThreadIOSentinel::current_default()))
+    ).clone();
+}
```

Comment:
How about something like this:
```
lazy_static! {
  static ref THREAD_IO_SENTINEL: Arc<ThreadLocal<ThreadIOSentinel>> = ...
}
struct AtomicIOBytes (AtomicUsize, AtomicUsize);
struct ThreadIOSentinel {
    id: ThreadID,
    last_bytes: AtomicIOBytes ,
    bytes: [AtomicIOBytes; IOType::COUNT],
}
```
This way we can use `THREAD_IO_SENTINEL.iter()` to implement `fetch_all_thread_io_bytes`.

Reply:
Sorry for leave so long!
I have tried this but I don't know how to deal with `io_type`. I have pushed a new commit which only `io_type` in mutex.

Reply:
Ok, maybe we can split `ThreadIOSentinel` into two parts, only one of them needs to be `Send` and iterable: 
```
thread_local! {
    THREAD_IO_PRIVATE: ThreadIOPrivate::new();
}
struct ThreadIOPrivate {
  io_type: IOType,
  last_bytes: IOBytes,
}

lazy_static! {
  static ref THREAD_IO_SENTINEL: Arc<ThreadLocal<ThreadIOSentinel>> = ...
}
struct ThreadIOSentinel {
    id: ThreadID,
    bytes: [AtomicIOBytes; IOType::COUNT],
}
```